### PR TITLE
Documentation

### LINKED ISSUE #81: Better docs
With the upcoming major release, the package is now probably feature-rich enough that it makes sense to start thinking about moving to full-featured documentation using sphinx and/or readthedocs.

### PR DESCRIPTION
**Pull request by camenpihor**
Fixes #81 

# Notes on reviewing

- **The diff count shows that this PR is large, but its quite misleading**. A lot of the line count is either from 
    - autogenerated files (conf.py, Make.bat, and Makefile) -- out of these one should only review conf.py if you're familiar with sphinx
    - ipython notebook diffs. I only updated slight wording and the number of octothorpes (#) in the headers, did not re-run any cells
- all code in `bambi/` are docstring related. I did not add or remove any documentation, but there were some spacing problems (which cause sphinx to have a hard time)
    - the only change of note here is that I changed some docstrings in `bambi/diagnostics.py` from the numpy convention to the Google convention, since most of the library uses the Google convention
- I would pretty much only look at the `.rst` file changes
- Happy to hear any thoughts people have whether it be reversions, continued code changes, or to back up any decisions I made!

# Changes

- Incorporates more information from the README.md
- Adds example notebooks
    - Made the `examples/` directory a symbolic link to the notebooks which are now housed in the documentation (similar to PyMC3)
    - Edited the headers in the notebooks since they are read in by sphinx (mostly this is making one top-level header `#` and making the other headers either `##` or `###`
    - No other edits to notebooks were made (mention this since notebooks are hard to review)
- API Reference now includes all classes, methods and functions from `models/`, `priors/`, `results/`, and `diagnostics`. 

# To run the docs locally

0. Grab this branch, and enter the root directory of this repository
1. enter an python3 virtual env of your choice (I used `conda create python=3.6`)
2. install the library `python setup.py install`
3. install requirements `pip install -r requirements-dev.txt`
4. enter the docs directory `cd docs/`
5. run the build `make html`
6. open the generated html in browser `open _build/html/index.html` (from still within the docs/ directory

# Next steps

- hosting on readthedocs? 
- If there is any custom CSS or images that people want on the docs?

### CODE REVIEW

**Comment by aloctavodia** [comment]
This is fantastic! I will try to build it locally tomorrow and provide some feedback. What about using github pages vs readthedocs?

**Comment by ColCarroll** [comment]
I haven't reviewed this, but I have my personal server set up to host docs from other projects. Everything built very cleanly (thanks @camenpihor for the instructions) and is hosted here: https://pymc3.colindcarroll.com/

**Comment by ColCarroll** [comment]
Only thing I could find was a link on the front page that probably should go to the CONTRIBUTING.md page in github: 
image

Also, now's probably the best time to make a bold, unpopular choice for a theme: https://sphinx-themes.org/ I vote for alabaster, namako, or materialdesign

Please ping me if you update and want a rebuild.

**Comment by camenpihor** [comment]
@ColCarroll Thanks !! Can you post a link to where they are hosted, I couldn't find them on your site, but thanks so much for hosting them there for us to review :) 

@aloctavodia I am not too familiar with where open source holds their docs, so I have no preference for github pages or readthedocs. Super happy to use github pages! Is that something that I can set up? Or does a maintainer have to do it?

@ColCarroll I'll check those themes out and make my vote, but I don't expect to have a strong preference and am happy to be overridden haha

**Comment by camenpihor** [comment]
Also, @ColCarroll Thanks for catching the contributing.md link mishap! I definitely meant to link there.. I can go through all of the links again!

**Comment by aloctavodia** [comment]
There are a [few ways](https://help.github.com/en/articles/configuring-a-publishing-source-for-github-pages) to set up github pages, I guess the easiest way is to create a branch named `gh-pages`.

**Comment by ColCarroll** [comment]
Er, it should be just at https://pymc3.colindcarroll.com/

File: docs/index.rst [hunk 1]
```diff
@@ -1,22 +1,92 @@
...
-.. toctree::
-   :maxdepth: 2
+.. image:: https://travis-ci.org/bambinos/bambi.svg?branch=master
+    :target: https://travis-ci.org/bambinos/bambi
+
+.. image:: https://coveralls.io/repos/github/bambinos/bambi/badge.svg?branch=master
+    :target: https://coveralls.io/github/bambinos/bambi?branch=master
+
+
+Bambi is a high-level Bayesian model-building interface written in Python. It's built on top of the `PyMC3 <https://github.com/pymc-devs/pymc3>`_ probabilistic programming framework, and is designed to make it extremely easy to fit mixed-effects models common in social sciences settings using a Bayesian approach.
```
Comment (aloctavodia) [inline docs/index.rst]: We should rephrase this to something like.

Bambi is a high-level Bayesian model-building interface written in Python. It works with two probabilistic programming frameworks, `PyMC3 <https://docs.pymc.io/>` or `PyStan <https://pystan.readthedocs.io/en/latest/>`, and is designed to make it extremely easy to fit Bayesian mixed-effects models common in biology, social sciences and other disciplines.

File: docs/index.rst [hunk 2]
```diff
@@ -1,22 +1,92 @@
...
+
+New Features
+============
+
+Bambi version ``0.1`` will be the final version supporting Python 2, but look forward to the forthcoming Bambi version ``0.2``!
```
Comment (aloctavodia) [inline docs/index.rst]: 0.1 --> 0.1.2

**Comment by aloctavodia** [comment]
I like Nameko or Alabaster. I also suggested two small changes, I will be happy to merge this after those comments are addressed.

File: docs/index.rst (hunk 1 above)
Comment (camenpihor) [reply docs/index.rst]: Sure! I think I just stole that from the README.md, should I update the readme too?

File: docs/index.rst (hunk 2 above)
Comment (camenpihor) [reply docs/index.rst]: Nice! Will change :)

**Comment by camenpihor** [comment]
@aloctavodia Thanks! I'll try to get to these tonight!

**Comment by camenpihor** [comment]
@aloctavodia It seems like I would need admin privileges on the repo in order to set up github pages :/ 

https://help.github.com/en/articles/configuring-a-publishing-source-for-github-pages#publishing-your-github-pages-site-from-a-docs-folder-on-your-master-branch

In particular, I need access to the repository's settings. 

I'm happy to learn about how to do it, and provide instructions to someone, if that would help? (I don't know who has admin privileges). Also happy to just let someone who has access do it haha

**Comment by camenpihor** [comment]
@aloctavodia Your suggestions have been addressed!
@ColCarroll I updated the docs, if you wanted to update the docs hosted on your site? (Thanks again!!)

# New Updates

* Alabaster theme
* Examples page now just contains the names of the pages to simplify the page
* Python code in the docs have been reformatted to fit in the page so that the viewer doesn't have to side-scroll

**Comment by ColCarroll** [comment]
Updated: https://pymc3.colindcarroll.com/

**Comment by twiecki** [comment]
I think I created it

**Comment by ColCarroll** [comment]
https://bambinos.github.io/bambi/ 

Looks like it is being served from the readme. In arviz, we use a travis job to build and [deploy the docs](https://github.com/arviz-devs/arviz/blob/master/scripts/deploy_docs.sh). It requires force pushing to the `gh-pages` branch, which requires giving travis a (secret) token: [instructions](https://github.com/Syntaf/travis-sphinx). The token would have to be added by an admin. I can help set up the travis stuff if that's of interest.

(we should merge this before doing any of that)

**Comment by aloctavodia** [comment]
ohhh! So it was always there?!

**Comment by camenpihor** [comment]
@aloctavodia Are there any other issues that you think would be nice/good for me to tackle?

**Comment by aloctavodia** [comment]
Not sure, I think we should try to do a release, the last one supporting Python 2.x and then move to remove python2 compatibility, integrate bambi with ArviZ. I would like to hear the opinion of others specially @tyarkoni.

**Comment by twiecki** [comment]
live at https://bambinos.github.io/bambi/
